### ISSUE TITLE
[Bug] MySQL Error when starting server

### ISSUE THREAD

**Issue by Yurinann** [Revivalo/DailyRewards#8]
```
[17:53:30 ERROR]: Failed to load driver class com.mysql.cj.jdbc.Driver from HikariConfig class classloader org.bukkit.plugin.java.PluginClassLoader@19ed928c
[17:53:30 ERROR]: Error occurred while enabling DailyRewards v1.6.3 (Is it up to date?)
java.lang.RuntimeException: Failed to load driver class com.mysql.cj.jdbc.Driver in either of HikariConfig class loader or Thread context classloader
        at com.zaxxer.hikari.HikariConfig.setDriverClassName(HikariConfig.java:491) ~[?:?]
        at cz.revivalo.dailyrewards.managers.database.MySQLManager.init(MySQLManager.java:37) ~[?:?]
        at cz.revivalo.dailyrewards.DailyRewards.onEnable(DailyRewards.java:83) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:321) ~[spigot.jar:git-iSpigot-PaperEdition]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:322) ~[spigot.jar:git-iSpigot-PaperEdition]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:389) ~[spigot.jar:git-iSpigot-PaperEdition]
        at org.bukkit.craftbukkit.v1_8_R3.CraftServer.loadPlugin(SourceFile:439) ~[spigot.jar:git-iSpigot-PaperEdition]
        at org.bukkit.craftbukkit.v1_8_R3.CraftServer.enablePlugins(SourceFile:398) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.MinecraftServer.s(SourceFile:505) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.MinecraftServer.k(SourceFile:430) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.MinecraftServer.a(SourceFile:398) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.DedicatedServer.init(SourceFile:388) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.MinecraftServer.run(SourceFile:728) ~[spigot.jar:git-iSpigot-PaperEdition]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

Server Version: ImanitySpigot 2022.11.1 LTS BUILD 3 with Java11

**Comment by Revivalo** [comment]
Can you try the DailyReward-DEVELOPMENT_BUILD from https://mega.nz/folder/AEB3SZwI#r8KX4i37tjcqIrFKYnQP7w

**Comment by Yurinann** [comment]
of course i will try it later



---Original---
From: ***@***.***&gt;
Date: Fri, Feb 10, 2023 05:14 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Revivalo/DailyRewards] [Bug] MySQL Error when starting server(Issue #8)




 
Can you try the DailyReward-DEVELOPMENT_BUILD from https://mega.nz/folder/AEB3SZwI#r8KX4i37tjcqIrFKYnQP7w

**Comment by Yurinann** [comment]
```
[20:30:35 ERROR]: Failed to load driver class com.mysql.cj.jdbc.Driver from HikariConfig class classloader org.bukkit.plugin.java.PluginClassLoader@5ce24378
[20:30:35 ERROR]: Error occurred while enabling DailyRewards v1.6.3 (Is it up to date?)
java.lang.RuntimeException: Failed to load driver class com.mysql.cj.jdbc.Driver in either of HikariConfig class loader or Thread context classloader
        at com.zaxxer.hikari.HikariConfig.setDriverClassName(HikariConfig.java:491) ~[?:?]
        at cz.revivalo.dailyrewards.managers.database.MySQLManager.init(MySQLManager.java:36) ~[?:?]
        at cz.revivalo.dailyrewards.DailyRewardsPlugin.onEnable(DailyRewardsPlugin.java:82) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:321) ~[spigot.jar:git-iSpigot-PaperEdition]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:322) ~[spigot.jar:git-iSpigot-PaperEdition]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:389) ~[spigot.jar:git-iSpigot-PaperEdition]
        at org.bukkit.craftbukkit.v1_8_R3.CraftServer.loadPlugin(SourceFile:439) ~[spigot.jar:git-iSpigot-PaperEdition]
        at org.bukkit.craftbukkit.v1_8_R3.CraftServer.enablePlugins(SourceFile:398) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.MinecraftServer.s(SourceFile:505) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.MinecraftServer.k(SourceFile:430) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.MinecraftServer.a(SourceFile:398) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.DedicatedServer.init(SourceFile:388) ~[spigot.jar:git-iSpigot-PaperEdition]
        at net.minecraft.server.v1_8_R3.MinecraftServer.run(SourceFile:728) ~[spigot.jar:git-iSpigot-PaperEdition]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

**Comment by Yurinann** [comment]
maybe we should add a config setting to custom database driver class because in MySQL5 it is "com.mysql.jdbc.Driver" and in MySQL8 it is "com.mysql.cj.jdbc.Driver"

**Comment by Revivalo** [comment]
Just noticed you are using 1.8 version.
This plugin support only 1.8.8 =< versions.

**Comment by Yurinann** [comment]
https://github.com/Imanity-Software/ImanitySpigot3-git

I use this version -- a 1.8.8 taco fork

anyway i will test with original 1.8.8 spigot later

**Comment by Yurinann** [comment]
@Revivalo Hi! I'm very sorry for the long delay in giving feedback because I was not able to test the environment before due to life changes during this time.

However, in my tests, this error is still reported. I guess this is because of the low version of jdbc driver? I wonder if it is working properly in your case?

BTW: I am using version 1.6.7 and have tested on both ImanitySpigot and paper-1.12.2-1620.

```
[17:27:54 ERROR]: [groupid.artifact.hikari.HikariConfig] Failed to load driver class com.mysql.cj.jdbc.Driver from HikariConfig class classloader org.bukkit.plugin.java.PluginClassLoader@c752d4b
[17:27:54 ERROR]: Error occurred while enabling DailyRewards v1.6.7 (Is it up to date?)
java.lang.RuntimeException: Failed to load driver class com.mysql.cj.jdbc.Driver in either of HikariConfig class loader or Thread context classloader
        at groupid.artifact.hikari.HikariConfig.setDriverClassName(HikariConfig.java:491) ~[?:?]
        at dev.revivalo.dailyrewards.managers.database.MySQLManager.init(MySQLManager.java:37) ~[?:?]
        at dev.revivalo.dailyrewards.DailyRewardsPlugin.onEnable(DailyRewardsPlugin.java:73) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.12.2.jar:git-Paper-1620]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:316) ~[patched_1.12.2.jar:git-Paper-1620]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) ~[patched_1.12.2.jar:git-Paper-1620]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugin(CraftServer.java:395) ~[patched_1.12.2.jar:git-Paper-1620]
        at org.bukkit.craftbukkit.v1_12_R1.CraftServer.enablePlugins(CraftServer.java:344) ~[patched_1.12.2.jar:git-Paper-1620]
        at net.minecraft.server.v1_12_R1.MinecraftServer.t(MinecraftServer.java:442) ~[patched_1.12.2.jar:git-Paper-1620]
        at net.minecraft.server.v1_12_R1.MinecraftServer.l(MinecraftServer.java:403) ~[patched_1.12.2.jar:git-Paper-1620]
        at net.minecraft.server.v1_12_R1.MinecraftServer.a(MinecraftServer.java:341) ~[patched_1.12.2.jar:git-Paper-1620]
        at net.minecraft.server.v1_12_R1.DedicatedServer.init(DedicatedServer.java:289) ~[patched_1.12.2.jar:git-Paper-1620]
        at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:616) ~[patched_1.12.2.jar:git-Paper-1620]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_362]
```

**Comment by Revivalo** [comment]
Fixed in 1.6.8 version.

**Comment by Yurinann** [comment]
In fact, the current approach leads to a situation where a database incompatibility related error occurs on the low version server with a local MySQL version of 8. Now judging the driver path based on version may not be reasonable enough, I'm trying to find a way to be perfectly compatible with all cases (but in fact I haven't found it yet).
